worldedit-core/src/main/java/com/sk89q/worldedit/function/pattern/RandomPattern.java

```diff
@@ -52,6 +52,19 @@ public RandomPattern(SimpleRandom random) {
         this.random = random;
     }
 
+    /**
+     * Create a random pattern from an existing one but with a different random.
+     *
+     * @param random the new random to use.
+     * @param parent the existing random pattern.
+     */
+    public RandomPattern(SimpleRandom random, RandomPattern parent) {
```

Comment:
We either need this one or make the random non-final, reassign it and recreate the RandomCollection with it

Reply:
Do whatever is easier here.